phoenix-core/src/main/java/org/apache/phoenix/hbase/index/IndexRegionObserver.java

```diff
@@ -1142,7 +1143,7 @@ public void preBatchMutateWithExceptions(ObserverContext<RegionCoprocessorEnviro
         identifyIndexMaintainerTypes(indexMetaData, context);
         identifyMutationTypes(miniBatchOp, context);
         context.populateOriginalMutations(miniBatchOp);
-
+        VerifyLastDDLTimestamp.verifyLastDDLTimestamp(miniBatchOp, c.getEnvironment());
```

Comment:
@shahrs87 So for indexes on non immutable tables we generate mutations on the server. I think those mutations will also need the last ddl timestamp attribute.

Reply:
The ddl timestamps are already verified when we create the mutations for the base table. When we upsert into base table, we validate the last ddl timestamp for base table as well as all the indexes.